src/advanced-types/type-bounds.md

```diff
@@ -1 +1,55 @@
 # Type Bounds
+When we express a [subtype-supertype relationship](/advanced-types/subtyping.html) by writing `T <: U`, then we say that `T` is *bounded* by `U`. Lets explore *type bounds* in more detail.
+
+## The `Any` and `None` types
+All types in Motoko are bounded by a special type, namely the `Any` type. This type is the *supertype* of all types and thus all types are a *subtype* of the `Any` type. We may refer to it as the *top type*. A variable in Motoko can be of type `Any`.
+
+Another special type in Motoko is the `None` type. This type is the *subtype* of all types and thus all types are a *supertype* of `None`. No value in Motoko can have the `None` type. 
+
+For any type `T` in Motoko, the following *type bounds* hold:  
+```motoko
+None <: T
+
+T <: Any
+```
+
+## Type bounds in functions
+Consider the following *function signature*:
+```motoko
+func makeNat<T <: Number>(x : T) : Natural
+```
+
+It's a [generic function](/advanced-types/generics.html) that specifies a *type bound* `Number` for its generic type parameter `T`. This is expressed as `<T <: Number>`. 
+
+The function takes an argument of generic bounded type `T` and returns a value of type `Natural`. The function is meant to take a general kind of number and process it into a `Nat`. This is sometimes called *type casting*.  
```

Comment:
```suggestion
The function takes an argument of generic bounded type `T` and returns a value of type `Natural`. The function is meant to take a general kind of number and process it into a `Nat`. 
```
Type-casting (in OO languages) typically doesn't change the value, just cast it up to a supertype (without any change) or do a checked downcast to a more precise subtype. You could call what you are doing here a more generic *conversion*.